src/SecurityManager.js

```diff
@@ -78,17 +103,21 @@ async function getSecretStorageKey({ keys: keyInfos }, ssssItemName) {
         return [keyId, secretStorageKeys[keyId]];
     }
 
-    const inputToKey = async ({ passphrase, recoveryKey }) => {
-        if (passphrase) {
-            return deriveKey(
-                passphrase,
-                keyInfo.passphrase.salt,
-                keyInfo.passphrase.iterations,
-            );
-        } else {
-            return decodeRecoveryKey(recoveryKey);
-        }
-    };
+    // if we dehydrated a device, see if that key works for SSSS
+    if (dehydrationInfo.key) {
+        try {
+            const key = dehydrationInfo.key;
+            if (await MatrixClientPeg.get().checkSecretStorageKey(key, keyInfo)) {
```

Comment:
For this to work, doesn't it mean there would need to be a step that attempts creating 4S with the dehydration key as the 4S key? Is that future work, or does some change here cause that to happen and I just haven't spotted it yet...? 🤔

Reply:
For now, we're not going to create 4S using the dehydration key, but we will unlock 4S using the dehydration key if it matches.

Reply:
Okay, but then... why do we want to do that, since we know that at the moment, it will always fail since 4S wasn't created with the dehydration key?

It seems like we would only want to start trying dehydration as a 4S key when there's some reason to believe it would work.

Reply:
Sorry, I missed this question earlier.

Even though we don't create 4S using the dehydration key, we do dehydrate the device using the 4S key, so they keys will be the same.